src/sinol_make/commands/run/__init__.py

```diff
@@ -562,11 +644,33 @@ def validate_expected_scores(self, results):
                         removed_groups.add(group[0])
             elif type == "change":
                 if field[1] == "expected": # Results for at least one group has changed
-                    solution = field[0]
-                    group = field[2]
-                    old_result = change[0]
-                    result = change[1]
-                    changes.append(ResultChange(solution, group, old_result, result))
+                    if isinstance(change[0], str) and isinstance(change[1], dict):
+                        changes.append(PointsChange(
+                            solution=field[0],
+                            group=field[2],
+                            old_points=self.scores[field[2]],
+                            new_points=change[1]["points"]
+                        ))
+                    elif isinstance(change[0], dict) and isinstance(change[1], str):
+                        changes.append(PointsChange(
+                            solution=field[0],
+                            group=field[2],
+                            old_points=change[0]["points"],
+                            new_points=self.scores[field[2]]
+                        ))
+                    elif isinstance(change[0], dict) and isinstance(change[1], dict):
+                        changes.append(PointsChange(
+                            solution=field[0],
+                            group=field[2],
+                            old_points=change[0]["points"],
+                            new_points=change[1]["points"]
+                        ))
+                    else:
+                        solution = field[0]
+                        group = field[2]
+                        old_result = change[0]
+                        result = change[1]
+                        changes.append(ResultChange(solution, group, old_result, result))
```

Comment:
Inconsistent styles in the branches. Also, the `solution` and `group` is always the same, so maybe it's possible to shorten the current code?

Reply:
Refactored c3c7c62958dfcbbee405ba26d7d4269907a695a4